core/src/main/kotlin/org/evomaster/core/output/dto/DtoWriter.kt

```diff
@@ -0,0 +1,90 @@
+package org.evomaster.core.output.dto
+
+import org.evomaster.core.output.OutputFormat
+import org.evomaster.core.output.TestWriterUtils
+import org.evomaster.core.problem.rest.param.BodyParam
+import org.evomaster.core.search.action.Action
+import org.evomaster.core.search.gene.BooleanGene
+import org.evomaster.core.search.gene.Gene
+import org.evomaster.core.search.gene.ObjectGene
+import org.evomaster.core.search.gene.datetime.DateGene
+import org.evomaster.core.search.gene.datetime.TimeGene
+import org.evomaster.core.search.gene.numeric.DoubleGene
+import org.evomaster.core.search.gene.numeric.FloatGene
+import org.evomaster.core.search.gene.numeric.IntegerGene
+import org.evomaster.core.search.gene.numeric.LongGene
+import org.evomaster.core.search.gene.string.Base64StringGene
+import org.evomaster.core.search.gene.string.StringGene
+import org.evomaster.core.search.gene.utils.GeneUtils
+import org.evomaster.core.utils.StringUtils
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import java.nio.file.Path
+
+object DtoWriter {
+
+    private val log: Logger = LoggerFactory.getLogger(DtoWriter::class.java)
+    private val dtoCollector: MutableMap<String, DtoClass> = mutableMapOf()
+
+    fun write(testSuitePath: Path, outputFormat: OutputFormat, actionDefinitions: List<Action>) {
+        getDtos(actionDefinitions)
+        dtoCollector.forEach {
+            JavaDtoWriter.write(testSuitePath, outputFormat, it.value)
+        }
+    }
+
+    private fun getDtos(actionDefinitions: List<Action>) {
+        actionDefinitions.forEach { action ->
+            action.getViewOfChildren().forEach { child ->
```

Comment:
instead of a `forEach` with a nested `if`, couldn't just use a `find`?

Reply:
I'm guessing there can be only one `BodyParam` per action, so yes, in that case it would definitely make much more sense to use `find`